### ISSUE TITLE
Undefined returning and errors

### ISSUE THREAD

**Issue by antiss11** [ynhhoJ/flibusta-api#35]
The problem is `null` returns in 2 cases: when error has place and no books in results. So there is no ways to separate errors and empty search results.

**Comment by ynhhoJ** [comment]
@antiss11 , can you provide some examples?

**Comment by antiss11** [comment]
Ok, i was wrong - in 2 cases returns `undefined`.

I've got next code:
```js
const flibustaApi = new FlibustaAPI(ORIGIN, { httpAgent });
const result = await flibustaApi.getBooksByNameFromOpds(bookTitle);
console.log(result);
```

And when I try to find book with title `12345` with active `tor` (or any proxy) service result is `undefined`.
After I stop `tor` and try to run code, I've got next output:
![2022-09-15-193139_560x37_scrot](https://user-images.githubusercontent.com/14955003/190458613-551aca6c-9284-402e-b421-086120e7ea3e.png)

**Comment by antiss11** [comment]
![2022-09-15-200527_773x136_scrot](https://user-images.githubusercontent.com/14955003/190466961-daffe194-f497-4ad5-9c7a-fe3206dbccef.png)

First part - with `tor`, second - without.
I think that the way you return errors isn't good. Ok, let it be, I'll try to do something later.

**Comment by ynhhoJ** [comment]
>  I think that the way you return errors isn't good. Ok, let it be, I'll try to do something later.

Yeah, i'll change that part soon. Wait for `0.4.1`!

**Comment by ynhhoJ** [comment]
Just released `0.4.1`.

Now on execution of code
```js
const flibustaApi = new FlibustaAPI(ORIGIN, { httpAgent });
const result = await flibustaApi.getBooksByNameFromOpds(bookTitle);
console.log(result);
```

Should log an error

**Comment by antiss11** [comment]
Works as intended. Good job!
